tools/ReleaseProcess/UploadArtifactsToGitHub.ps1

```diff
@@ -0,0 +1,444 @@
+## Copyright (c) Microsoft. All rights reserved.
+## Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+<#
+    .SYNOPSIS
+        This function downloads and prepares the Iot Edgelet artifacts from DevOps pipeline for a release. 
+        The final artifacts are collected under "<WorkDir>/output" path.
+    .NOTES
+        Requires Azure CLI to be login
+    .DESCRIPTION
+        1) Download edgelet artifacts from a given BuildId from a DevOps pipeline
+        2) Extract the *.zip artifacts into their own directories
+        3) Traverse the directories and rename a relavant artifact into the following format: 
+              .DEB _________________ {component}_{version}_{os}_{arch}.deb 
+              .RPM _________________ {component}-{version}.{os}.{arch}.rpm  (No name change)
+        4) Move the renamed artifacts to a staging directory at "<WorkDir>/output"
+    .PARAMETER BuildId
+        DevOps build ID for the downloading artifacts
+    .PARAMETER WorkDir
+        Absolute path of the working directory
+#>
+function Prepare-DevOps-Artifacts
+{
+    [CmdletBinding()]
+    param (
+        <# 
+        DevOps build ID for the downloading artifacts
+        #>
+        [Parameter(Mandatory)]
+        [string]
+        $BuildId,
+
+        <# 
+        Absolute path of the working directory
+        #>
+        [Parameter(Mandatory)]
+        [string]
+        $WorkDir
+    )
+
+    # Assume Az CLI is installed & logged in.
+    $pat = $(az keyvault secret show -n IotEdge1-PAT-msazure --vault-name edgebuildkv | ConvertFrom-Json)
+    $pat = $pat.value;
+
+    $encoded64Pat = [Convert]::ToBase64String([System.Text.Encoding]::UTF8.GetBytes(':'+$pat))
+    $header = @{
+        "Authorization" = "Basic $encoded64Pat"
+    }
+
+    # Get all the artifact names
+    $response = $(Invoke-WebRequest -Uri "https://dev.azure.com/msazure/One/_apis/build/builds/$BuildId/artifacts?api-version=6.0" -Headers $header)
```

Comment:
Yeah... I hear you. I can make the API calls more generic but I'm not entirely sure how useful it will be for any other teams given that we currently want to download & publish the artifacts (with hard-coded names) to our repo. 

If there is a certain use case that we want to support, I think a Cx can open a feature request against the script.